WordPress/UITestsFoundation/Screens/Editor/BlockEditorScreen.swift

```diff
@@ -1,42 +1,37 @@
+import ScreenObject
 import XCTest
 
-public class BlockEditorScreen: BaseScreen {
+public class BlockEditorScreen: ScreenObject {
 
-    // Navigation bar
-    let editorNavBar = XCUIApplication().navigationBars["Gutenberg Editor Navigation Bar"]
-    let editorCloseButton = XCUIApplication().navigationBars["Gutenberg Editor Navigation Bar"].buttons["Close"]
-    let publishButton = XCUIApplication().buttons["Publish"]
-    let publishNowButton = XCUIApplication().buttons["Publish Now"]
-    let moreButton = XCUIApplication().buttons["more_post_options"]
+    let editorCloseButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.navigationBars["Gutenberg Editor Navigation Bar"].buttons["Close"]
+    }
 
-    // Editor area
-    // Title
-    let titleView = XCUIApplication().otherElements["Post title. Empty"].firstMatch // Uses a localized string
-    // Paragraph block
-    let paragraphView = XCUIApplication().otherElements["Paragraph Block. Row 1. Empty"].textViews.element(boundBy: 0)
-    // Image block
-    let imagePlaceholder = XCUIApplication().buttons["Image block. Empty"] // Uses a localized string
+    var editorCloseButton: XCUIElement { editorCloseButtonGetter(app) }
 
-    // Toolbar
-    let addBlockButton = XCUIApplication().buttons["add-block-button"] // Uses a testID
+    let addBlockButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.buttons["add-block-button"] // Uses a testID
+    }
 
-    // Action sheets
-    let actionSheet = XCUIApplication().sheets.element(boundBy: 0)
-    let imageDeviceButton = XCUIApplication().sheets.buttons["Choose from device"] // Uses a localized string
-    let discardButton = XCUIApplication().buttons["Discard"] // Uses a localized string
-    let postSettingsButton = XCUIApplication().sheets.buttons["Post Settings"] // Uses a localized string
-    let keepEditingButton = XCUIApplication().sheets.buttons["Keep Editing"] // Uses a localized string
+    var addBlockButton: XCUIElement { addBlockButtonGetter(app) }
 
-    public init() {
-        // Check addBlockButton element to ensure block editor is fully loaded
-        super.init(element: addBlockButton)
+    public init(app: XCUIApplication = XCUIApplication()) throws {
+        // The block editor has _many_ elements but most are loaded on-demand. To verify the screen
+        // is loaded, we rely only on the button to add a new block and on the navigation bar we
+        // expect to encase the screen.
```

Comment:
This is also the reason for all the red lines above. Instead of instantiating the elements at `init` time, I've updated the code to use `ScreenObject`'s `app` property and query it at runtime when a method that needs an element is called.

Also, a few of the elements were unused 🧹 🧹 🧹